pytest/tests/loadtest/locust/common/base.py

```diff
@@ -0,0 +1,327 @@
+import base64
+import json
+import base58
+import ctypes
+import locust
+import logging
+import multiprocessing
+import pathlib
+import random
+import requests
+import sys
+import time
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[4] / 'lib'))
+
+import cluster
+import transaction
+import key
+import mocknet_helpers
+import utils
+
+from locust import HttpUser, events
+from configured_logger import new_logger
+
+DEFAULT_TRANSACTION_TTL_SECONDS = 20
+logger = new_logger(level=logging.WARN)
+
+
+class Account:
+
+    def __init__(self, key):
+        self.key = key
+        self.nonce = multiprocessing.Value(ctypes.c_ulong, 0)
+
+    def refresh_nonce(self, node):
+        with self.nonce.get_lock():
+            self.nonce.value = mocknet_helpers.get_nonce_for_key(
+                self.key,
+                addr=node.rpc_addr()[0],
+                port=node.rpc_addr()[1],
+                logger=logger,
+            )
+
+    def use_nonce(self):
+        with self.nonce.get_lock():
+            new_nonce = self.nonce.value + 1
+            self.nonce.value = new_nonce
+            return new_nonce
+
+
+class Transaction:
+    """
+    A transaction future.
+    """
+
+    ID = 0
+
+    def __init__(self):
+        self.id = Transaction.ID
+        Transaction.ID += 1
+
+        # The transaction id hash
+        #
+        # str if the transaction has been submitted and may eventually conclude.
+        # FIXME: this is currently not set in some cases
+        self.transaction_id = None
+
+    def sign_and_deser(self, block_hash):
+        """
+        Each transaction class is supposed to define this method to deserialize and
```

Comment:
I don't understand why this method claims to "deserialize" a transaction, I think it should be "serialize" instead given that it returns a raw message and a typical implementation actually does serialization:
https://github.com/near/nearcore/blob/eb7a184f03790d23483a0949452bd21efa8c07f3/pytest/lib/transaction.py#LL183C12-L183C30

So the name should be `sign_and_serialize` or just `sign` if that's not confusing in this context?

Reply:
Oh yeah, stupid me :) Renamed to `sign_and_serialize`, I like the extra context to make it clear the result is binary and not a struct.